### PR TITLE
Handle invalid user

### PR DESCRIPTION
**Pull request by rullzer**
In case a user has been removed (LDAP) but is still visible on the
system. We should just catch and continue.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: lib/BackgroundJob/RetentionJob.php [hunk 1]
```diff
@@ -161,7 +161,11 @@ private function checkFileId($fileid) {
 
 		$mountPoint = array_shift($mountPoints);
 
-		$userFolder = $this->rootFolder->getUserFolder($mountPoint->getUser()->getUID());
+		try {
+			$userFolder = $this->rootFolder->getUserFolder($mountPoint->getUser()->getUID());
+		} catch (\Exception $e) {
+			throw new NotFoundException('Could not get user');
```
Comment (MorrisJobke) [inline lib/BackgroundJob/RetentionJob.php]: Please just log the exception.

**Comment by rullzer** [comment]
I'll try to fix the tests properly first... so we have green ci.

**Comment by rullzer** [comment]
Done

**Pull request merged**